Curator: revise header template part

jffng opened the pull request:
<!-- Thanks for contributing to our free themes! Please provide as much information as possible with your Pull Request by filling out the following - this helps make reviewing much quicker! -->

#### Changes proposed in this Pull Request:

This PR makes adjustments to the header part and associated styles.

Before | After
---- | ----
<img width="1775" alt="Screen Shot 2022-05-06 at 1 57 40 PM" src="https://user-images.githubusercontent.com/5375500/167190493-2550f301-694e-4023-9ee0-75b140125907.png"> | <img width="1783" alt="Screen Shot 2022-05-06 at 1 57 28 PM" src="https://user-images.githubusercontent.com/5375500/167190516-feb19c0d-9ee6-4d7f-a4fe-69cdb73bb04e.png">


I'm not sure why the color of the style title is not being correctly applied. 

#### Related issue(s):

Fixes #5925

jffng committed:
Revise to match design.

jffng committed:
Remove text decoration from links and make vertical spacing larger.

madhusudhand commented:
**Design:**

<img width="877" alt="image" src="https://user-images.githubusercontent.com/1935113/167342318-cbc9d120-0ffb-4822-9305-4ccfb0955aca.png">

**frontend:**

<img width="898" alt="image" src="https://user-images.githubusercontent.com/1935113/167342249-3223d062-c669-4ec5-8f12-6ad0e3d0cbbf.png">

**FSE:**

<img width="1149" alt="image" src="https://user-images.githubusercontent.com/1935113/167342380-f602ab1b-453c-401e-b1ed-23c940e9dfb9.png">

Following are the observations:

* site title color should be "foreground" (black)
* text is not vertically centered. (this is because of the 9px margin of separator block)
* separator block is not full width in FSE

MaggieCabrera commented on curator/parts/header.html:
```diff
@@ -1,29 +1,13 @@
+<div class="wp-block-group alignfull" style="padding-top:var(--wp--custom--gap--vertical);padding-bottom:var(--wp--custom--gap--vertical)"><!-- wp:site-title {"style":{"typography":{"fontSize":"1.375rem"}},"textColor":"foreground"} /-->
 
-</div>
+<!-- wp:navigation {"__unstableLocation":"primary","layout":{"type":"flex","setCascadingProperties":true,"justifyContent":"right","orientation":"horizontal"},"style":{"spacing":{"margin":{"top":"0"}}}} /--></div>
```
Do we need unstable location for a non universal theme? I don't see we've used it on archeo or remote for example

jffng replied:
Could it still be useful for someone switching to this theme from a non-block theme that has a primary navigation defined? 

If this is not a likely scenario or it doesn't work this way anymore, I think we can remove it.

jffng committed:
Remove margin from separator.

jffng committed:
Revert text decoration none.

jffng committed:
Make appearance tools available and adjust site title link color accordingly.

jffng commented:
> site title color should be "foreground" (black)

Fixed in https://github.com/Automattic/themes/pull/5980/commits/cd887fc740f64a20e166fd705e8dd60fceacd5ce

> text is not vertically centered. (this is because of the 9px margin of separator block)

Fixed in https://github.com/Automattic/themes/pull/5980/commits/9714bea0c80485e59f5f360c8857b908127f622b.

> separator block is not full width in FSE

This is due to a bug upstream. I opened a fix here: https://github.com/WordPress/gutenberg/pull/40937

jffng committed:
Styles for site title.

jffng committed:
Remove unstable location.

jffng commented:
This is ready for another review. I think I addressed everything except the separator full width issue, which needs to be resolved upstream.

I'm addressing the mobile menu styling in a separate PR: https://github.com/Automattic/themes/pull/5988

jffng committed:
Remove header.

jffng committed:
Add back EOL

beafialho commented:
The color of the Site title shows up correctly for me 👍 A few comments:

- The Site Title is underlined, and it should be underlined only on hover.

- I'm not sure if it's related to the header template part, but the single post should literally stick to the separator, without the white spacing:

<img width="1536" alt="Captura de ecrã 2022-05-27, às 11 17 08" src="https://user-images.githubusercontent.com/65220155/170680887-51913853-442d-4c56-ab16-bd7d4b86a23b.png">

- This doesn't look ideal. The stroke should at least be the same color. We'd have to add CSS to customize this as this is GB's default right?
<img width="392" alt="Captura de ecrã 2022-05-27, às 11 20 54" src="https://user-images.githubusercontent.com/65220155/170681626-88159783-1bd7-45f1-b712-a878a305cbea.png">

madhusudhand commented:
> The Site Title is underlined, and it should be underlined only on hover.

![curator-header-hover](https://user-images.githubusercontent.com/1935113/171567174-62a5bd13-f1d9-4a7e-a362-55e34d7ac7d3.gif)

> I'm not sure if it's related to the header template part, but the single post should literally stick to the separator, without the white spacing:

Issue is with Single post template. Header is occupying right height.

<img width="778" alt="image" src="https://user-images.githubusercontent.com/1935113/171567528-710d09ff-802c-411b-9f58-75f25e6b3886.png">

> This doesn't look ideal. The stroke should at least be the same color. We'd have to add CSS to customize this as this is GB's default right?

Right, it is GB default. We would need custom styles to override the behaviour. Following is the result with border color set to foreground color.

<img width="309" alt="image" src="https://user-images.githubusercontent.com/1935113/171569254-bf8af640-e7c1-4540-9c1b-54c8d48d7e49.png">

Is this looking fine?

madhusudhand committed:
fix nav menu border color

jffng committed:
Fix rebase.

jffng committed:
Remove margin from single template cover.

jffng commented:
> I'm not sure if it's related to the header template part, but the single post should literally stick to the separator, without the white spacing:

Fixed in https://github.com/Automattic/themes/pull/5980/commits/c1eedfd5bc3a473b289ca6a1203837422315e9a6

The rest of the feedback I think has been addressed, so I think this is in a good place to be brought in and we can iterate. If there's more feedback on the nav submenu we can address in a follow up. Thoughts?

jffng commented:
Going to bring this in since it will make other template PRs easier to review from a design perspective. Can address additional feedback in follow ups, thanks.
